cli/src/commit_templater.rs

```diff
@@ -620,6 +620,29 @@ fn builtin_commit_methods<'repo>() -> CommitTemplateBuildMethodFnMap<'repo, Comm
             Ok(L::wrap_boolean(out_property))
         },
     );
+    map.insert(
+        "contained_in",
+        |language, build_ctx, self_property, function| {
+            let [revset_node] = template_parser::expect_exact_arguments(function)?;
+
+            let revset =
+                template_builder::expect_plain_text_expression(language, build_ctx, revset_node)?
+                    .extract()
+                    .map_err(|err| {
+                        TemplateParseError::expression(
+                            "Failed to extract revset param",
+                            function.args_span,
+                        )
+                        .with_source(err.0)
+                    })?;
+            let revset = evaluate_user_revset(language, function.name_span, revset)?;
```

Comment:
`template_parser::expect_string_literal_with()` can be used for now. (see `builtin_timestamp_methods` for example.)

I'm thinking of replacing it to support dynamic evaluation, but I'm not too sure if that's a good (so haven't sent PR yet.)
```rust
if let Ok(_) = node.extract() {
    // static property fn (cached)
} else {
    // dynamic property fn
}
```

Reply:
ty, that's a lot nicer. 👍🏻